Sporadic query failures due to a deadlock in SqlServerClient.getTableProperties: "Transaction was deadlocked ... and has been chosen as the deadlock victim".
https://github.com/prestosql/presto/runs/1581864249
2020-12-19T15:49:54.7256766Z [ERROR] testStddevPushdown(io.prestosql.plugin.sqlserver.TestSqlServerIntegrationSmokeTest)  Time elapsed: 13.253 s  <<< FAILURE!
2020-12-19T15:49:54.7271020Z java.lang.RuntimeException: Unable to advance result set [statement:"SELECT data_compression_desc FROM sys.partitions p INNER JOIN sys.tables t ON p.object_id = t.object_id INNER JOIN sys.indexes i ON t.object_id = i.object_id WHERE SCHEMA_NAME(t.schema_id) = :schema AND t.name = :table_name AND i.type IN (0,1) AND i.data_space_id NOT IN (SELECT data_space_id FROM sys.partition_schemes)", arguments:{positional:{}, named:{schema:dbo,table_name:test_stddev_pushdown_gn0z4}, finder:[]}]
2020-12-19T15:49:54.7273851Z 	at io.prestosql.testing.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:115)
2020-12-19T15:49:54.7277668Z 	at io.prestosql.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:462)
2020-12-19T15:49:54.7282982Z 	at io.prestosql.sql.query.QueryAssertions$QueryAssert.verifyResultsWithPushdownDisabled(QueryAssertions.java:396)
2020-12-19T15:49:54.7287666Z 	at io.prestosql.sql.query.QueryAssertions$QueryAssert.isFullyPushedDown(QueryAssertions.java:346)
2020-12-19T15:49:54.7294040Z 	at io.prestosql.plugin.sqlserver.TestSqlServerIntegrationSmokeTest.testStddevPushdown(TestSqlServerIntegrationSmokeTest.java:191)
2020-12-19T15:49:54.7299413Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2020-12-19T15:49:54.7303649Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2020-12-19T15:49:54.7309009Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2020-12-19T15:49:54.7312634Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2020-12-19T15:49:54.7316747Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
2020-12-19T15:49:54.7320371Z 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
2020-12-19T15:49:54.7323848Z 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
2020-12-19T15:49:54.7328079Z 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
2020-12-19T15:49:54.7332967Z 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
2020-12-19T15:49:54.7337454Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
2020-12-19T15:49:54.7342284Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2020-12-19T15:49:54.7347196Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2020-12-19T15:49:54.7350901Z 	at java.base/java.lang.Thread.run(Thread.java:834)
2020-12-19T15:49:54.7363757Z Caused by: org.jdbi.v3.core.result.ResultSetException: Unable to advance result set [statement:"SELECT data_compression_desc FROM sys.partitions p INNER JOIN sys.tables t ON p.object_id = t.object_id INNER JOIN sys.indexes i ON t.object_id = i.object_id WHERE SCHEMA_NAME(t.schema_id) = :schema AND t.name = :table_name AND i.type IN (0,1) AND i.data_space_id NOT IN (SELECT data_space_id FROM sys.partition_schemes)", arguments:{positional:{}, named:{schema:dbo,table_name:test_stddev_pushdown_gn0z4}, finder:[]}]
2020-12-19T15:49:54.7370100Z 	at org.jdbi.v3.core.result.ResultSetResultIterator.safeNext(ResultSetResultIterator.java:108)
2020-12-19T15:49:54.7375451Z 	at org.jdbi.v3.core.result.ResultSetResultIterator.hasNext(ResultSetResultIterator.java:60)
2020-12-19T15:49:54.7380114Z 	at org.jdbi.v3.core.result.ResultIterable.findOne(ResultIterable.java:156)
2020-12-19T15:49:54.7406158Z 	at io.prestosql.plugin.sqlserver.SqlServerClient.getTableDataCompression(SqlServerClient.java:437)
2020-12-19T15:49:54.7409551Z 	at io.prestosql.plugin.sqlserver.SqlServerClient.getTableProperties(SqlServerClient.java:378)
2020-12-19T15:49:54.7413134Z 	at io.prestosql.plugin.jdbc.ForwardingJdbcClient.getTableProperties(ForwardingJdbcClient.java:288)
2020-12-19T15:49:54.7416761Z 	at io.prestosql.plugin.jdbc.jmx.StatisticsAwareJdbcClient.getTableProperties(StatisticsAwareJdbcClient.java:304)
2020-12-19T15:49:54.7420320Z 	at io.prestosql.plugin.jdbc.CachingJdbcClient.getTableProperties(CachingJdbcClient.java:365)
2020-12-19T15:49:54.7423854Z 	at io.prestosql.plugin.jdbc.CachingJdbcClient.getTableProperties(CachingJdbcClient.java:365)
2020-12-19T15:49:54.7426713Z 	at io.prestosql.plugin.jdbc.JdbcMetadata.getTableMetadata(JdbcMetadata.java:344)
2020-12-19T15:49:54.7429342Z 	at io.prestosql.metadata.MetadataManager.getTableMetadata(MetadataManager.java:508)
2020-12-19T15:49:54.7431916Z 	at io.prestosql.sql.planner.InputExtractor.createInput(InputExtractor.java:66)
2020-12-19T15:49:54.7436831Z 	at io.prestosql.sql.planner.InputExtractor$Visitor.processScan(InputExtractor.java:102)
2020-12-19T15:49:54.7438354Z 	at io.prestosql.sql.planner.InputExtractor$Visitor.visitTableScan(InputExtractor.java:84)
2020-12-19T15:49:54.7439900Z 	at io.prestosql.sql.planner.InputExtractor$Visitor.visitTableScan(InputExtractor.java:71)
2020-12-19T15:49:54.7441486Z 	at io.prestosql.sql.planner.plan.TableScanNode.accept(TableScanNode.java:143)
2020-12-19T15:49:54.7443489Z 	at io.prestosql.sql.planner.InputExtractor$Visitor.visitPlan(InputExtractor.java:109)
2020-12-19T15:49:54.7444925Z 	at io.prestosql.sql.planner.InputExtractor$Visitor.visitPlan(InputExtractor.java:71)
2020-12-19T15:49:54.7446639Z 	at io.prestosql.sql.planner.plan.PlanVisitor.visitAggregation(PlanVisitor.java:29)
2020-12-19T15:49:54.7448501Z 	at io.prestosql.sql.planner.plan.AggregationNode.accept(AggregationNode.java:203)
2020-12-19T15:49:54.7450225Z 	at io.prestosql.sql.planner.InputExtractor.lambda$extractInputs$0(InputExtractor.java:54)
2020-12-19T15:49:54.7451891Z 	at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:405)
2020-12-19T15:49:54.7453580Z 	at io.prestosql.sql.planner.InputExtractor.extractInputs(InputExtractor.java:54)
2020-12-19T15:49:54.7455417Z 	at io.prestosql.execution.SqlQueryExecution.doPlanQuery(SqlQueryExecution.java:457)
2020-12-19T15:49:54.7457231Z 	at io.prestosql.execution.SqlQueryExecution.planQuery(SqlQueryExecution.java:430)
2020-12-19T15:49:54.7458932Z 	at io.prestosql.execution.SqlQueryExecution.start(SqlQueryExecution.java:382)
2020-12-19T15:49:54.7460607Z 	at io.prestosql.execution.SqlQueryManager.createQuery(SqlQueryManager.java:237)
2020-12-19T15:49:54.7462722Z 	at io.prestosql.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
2020-12-19T15:49:54.7464832Z 	at io.prestosql.$gen.Presto_testversion____20201219_154259_200.run(Unknown Source)
2020-12-19T15:49:54.7466248Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2020-12-19T15:49:54.7467907Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2020-12-19T15:49:54.7469798Z 	at java.base/java.lang.Thread.run(Thread.java:834)
2020-12-19T15:49:54.7471276Z Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Transaction (Process ID 52) was deadlocked on lock resources with another process and has been chosen as the deadlock victim. Rerun the transaction.
2020-12-19T15:49:54.7475238Z 	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:254)
2020-12-19T15:49:54.7477578Z 	at com.microsoft.sqlserver.jdbc.SQLServerResultSet$FetchBuffer.nextRow(SQLServerResultSet.java:5378)
2020-12-19T15:49:54.7479795Z 	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.fetchBufferNext(SQLServerResultSet.java:1754)
2020-12-19T15:49:54.7482318Z 	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.next(SQLServerResultSet.java:1018)
2020-12-19T15:49:54.7486880Z 	at org.jdbi.v3.core.result.ResultSetResultIterator.safeNext(ResultSetResultIterator.java:106)
2020-12-19T15:49:54.7488169Z 	... 31 more

      